mlir/lib/Dialect/Rock/Transforms/BlockwiseGemmToThreadwise.cpp

```diff
@@ -777,11 +634,13 @@ LogicalResult ThreadwiseReadIntoRewritePattern::matchAndRewrite(
     if (srcAddrSpace == gpu::AddressSpace::Global) {
       Value loaded = b.create<GlobalLoadOp>(
           loc, loadType, buffer, loadLoop.getValidity(/*domain=*/0),
-          loadLoop.getLowerCoords(/*domain=*/0));
+          loadLoop.getLowerCoords(/*domain=*/0), needs64BitIdx);
       b.create<InBoundsStoreOp>(loc, loaded, dest,
                                 loadLoop.getLowerCoords(
                                     /*domain=*/1)[extraIdxCount]);
     } else {
+      assert(!needs64BitIdx &&
```

Comment:
Im not sure asserts are the right choice here... 
Cant a user trip over this ? or are they covered by verifiers ?